src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -577,7 +584,14 @@ public void onUpgradeComplete(final int newLevel)
         this.setHeight(wrapper.getHeight());
         this.setCorners(corners.getFirst().getFirst(), corners.getFirst().getSecond(), corners.getSecond().getFirst(), corners.getSecond().getSecond());
         this.isBuilt = true;
+
+        if (newLevel > getBuildingLevel())
+        {
+            spawnFireworksAtBuildingCorners(getTargetableArea(colony.getWorld()), colony, newLevel);
```

Comment:
Do not import the static method directly.
Use something like `FireworksUtils.spawnFireworksAtCornersOfAABB(...);`